WordPress/src/main/java/org/wordpress/android/ui/reader/discover/interests/ReaderInterestsViewModel.kt

```diff
@@ -37,15 +41,20 @@ class ReaderInterestsViewModel @Inject constructor(
     }
 
     private fun loadInterests() {
-        viewModelScope.launch {
-            val tagList = readerTagRepository.getInterests()
-            val currentUiState = uiState.value as UiState
-            updateUiState(
-                ContentLoadSuccessUiState(
-                    interestTagsUiState = transformToInterestsUiState(tagList),
-                    interestTags = tagList,
-                    doneBtnUiState = currentUiState.getDoneButtonState()
+        updateUiState(LoadingUiState)
+        if (networkUtils.isNetworkAvailable()) {
```

Comment:
I believe the VM doesn't/shouldn't even know where is the data retrieved from. Moreover, if we have some data in the cache I assume, we'll want to show them instead of showing a fullscreen error. Wdyt?

Reply:
You're right. Removed n/w condition here: 8a83960.